src/dune_engine/process.mli

```diff
@@ -83,6 +83,12 @@ val create_metadata :
    this behaviour by setting [set_temp_dir_when_running_actions = false]. *)
 val set_temp_dir_when_running_actions : bool ref
 
+(** [with_execution_context ~display ~f] executes [f] where:
+
+    - [display] is the display provider *)
+val with_execution_context :
+  display:Display.t -> f:(unit -> 'a Fiber.t) -> 'a Fiber.t
```

Comment:
This seems a pretty roundabout way of setting some parameters. There are a few issues:
* I find that this function lives in a pretty odd place. 
* It works by magic (setting some variable) rather than by changing types. Why not return something different instead of `Fiber.t`? Maybe use `Job.t` and even move it out into a separate module?
* Failing at runtime (via code errors) instead of compile time isn't great but that seems fixable (see the above comment).

Reply:
If it were to return `'a -> Job.t`, would this type be isomorphic to `Display.t -> 'a Fiber.t`? If so, is it really a big improvement?

Reply:
Compared to the current magic, yes!

Reply:
`Display.t -> 'a Fiber.t` is so much easier to reason about.

Reply:
Isn't it going to be a pair `'a Job.t = (Display.t, 'a Fiber.t)` though? Still easy to reason about. Just not sure why you suggested using a function instead.

Reply:
I guess i don't see the advantage of just passing `Display.t` then. Otherwise, we'll have to call `val run : 'a Job.t -> Display.t -> 'a Fiber.t` basically everywhere we use the functions in Process.

Reply:
With my definition of `'a Job.t` (above), I think you don't need to pass the display parameter.

(Having said that, just passing it everywhere seems fine too?)

Reply:
> Isn't it going to be a pair 'a Job.t = (Display.t, 'a Fiber.t) though? Still easy to reason about. Just not sure why you suggested using a function instead.

In the current implementation, the computation of `'a Fiber.t` depends on `Display.t`. It could be possible to refactor it though.

Reply:
Right. I think it's useful to distinguish "what to run" (which is captured by the fiber) from "how to run it" (which is captured by parameters like `Display.t`).

Reply:
Sure, that sounds reasonable. But no matter the representation one would still need the `run`  function I've mentioned inserted everywhere, right?

Reply:
Having to say how to execute a fiber at any place we actually execute a fiber seems fine. I guess that's going to make the code more explicit, but hopefully not painfully so.

Reply:
Yup. As I said, I don't mind passing the parameter everywhere. And I think this we can agree that having this Job.t type and passing a parameter is the same from the point of view of boilerplate anyways.

Reply:
Agreed!